imap_processing/idex/idex_packet_parser.py

```diff
@@ -0,0 +1,286 @@
+import logging
+
+import bitstring
+import numpy as np
+import xarray as xr
+from space_packet_parser import parser, xtcedef
+
+from imap_processing import packet_definition_directory
+
+TWENTY_MICROSECONDS = 20 * (10 ** (-6))
+
+
+class IDEXPacketParser:
+    """
+    This class encapsulates the decom work needed to decom a daily file of IDEX data
+    received from the POC.  The class is instantiated with a reference to a L0 file as
+    it exists on the local file system.
+
+    Parameters:
+        filename (str):  The path to the CDF file to read
+
+     Example:
+        >>> # Print out the data in a single packet
+        >>> from imap_processing.idex.idex_packet_parser import IDEXPacketParser
+        >>> l0_file = "imap_processing/idex/tests/imap_idex_l0_20230725_v01-00.pkts"
+        >>> l0_data = IDEXPacketParser(l0_file)
+        >>> print(l0_data.epochs)
+        >>> print(l0_data.data)
+    """
+
+    def __init__(self, idex_packet_file: str):
+        """
+        This function takes in a local l0 pkts file and performs all of the decom work
+        directly in __init__().
+
+        Parameters:
+            idex_packet_file (str):  The path and filename to the L0 file to read
+
+        Returns:
+            Populates {object}.data, where data is an xarray DataSet
+
+        Notes:
+            Currently assumes one L0 file will generate exactly one l1a file
+
+            There are a few things that must be determined in the future:
+            * high_sample_rate and low_sample_rate must be multiplied by
+              some scale factor that will be decided upon in the future
+            * IDEX_Trigger is a placeholder variable for now
+        """
+
+        idex_xtce = f"{packet_definition_directory}/idex_packet_definition.xml"
+        idex_definition = xtcedef.XtcePacketDefinition(xtce_document=idex_xtce)
+        idex_parser = parser.PacketParser(idex_definition)
+
+        idex_binary_data = bitstring.ConstBitStream(filename=idex_packet_file)
+        idex_packet_generator = idex_parser.generator(idex_binary_data)
+
+        self.coords = {}
+        self.data = {}
+        self.dust_events = {}
```

Comment:
Not used anywhere else, so just make it a local variable?
```suggestion
        dust_events = {}
```